otx/algorithms/action/adapters/mmaction/models/backbones/movinet.py

```diff
@@ -0,0 +1,1048 @@
+"""
+Code inspired by:
+https://github.com/Atze00/MoViNet-pytorch/blob/main/movinets/models.py
+"""
+
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+from collections import OrderedDict
+from typing import Any, Callable, Optional, Tuple, Union
+
+import torch
+import torch.nn.functional as F
+from einops import rearrange
+from mmaction.models.builder import BACKBONES
+from torch import Tensor, nn
+from torch.nn.modules.utils import _pair, _triple
+from yacs.config import CfgNode
+
+
+class CausalModule(nn.Module):
+    def __init__(self) -> None:
+        super().__init__()
+        self.activation = None
+
+    def reset_activation(self) -> None:
+        self.activation = None
+
+
+class TemporalCGAvgPool3D(CausalModule):
+    def __init__(
+        self,
+    ) -> None:
+        super().__init__()
+        self.n_cumulated_values = 0
+        self.register_forward_hook(self._detach_activation)
+
+    def forward(self, x: Tensor) -> Tensor:
+        input_shape = x.shape
+        device = x.device
+        cumulative_sum = torch.cumsum(x, dim=2)
+        if self.activation is None:
+            self.activation = cumulative_sum[:, :, -1:].clone()
+        else:
+            cumulative_sum += self.activation
+            self.activation = cumulative_sum[:, :, -1:].clone()
+        divisor = torch.arange(1, input_shape[2] + 1, device=device)[None, None, :, None, None].expand(x.shape)
+        x = cumulative_sum / (self.n_cumulated_values + divisor)
+        self.n_cumulated_values += input_shape[2]
+        return x
+
+    @staticmethod
+    def _detach_activation(module: CausalModule, input: Tensor, output: Tensor) -> None:
+        module.activation.detach_()
+
+    def reset_activation(self) -> None:
+        super().reset_activation()
```

Comment:
how about using self.reset_activation() instead of super().reset_activation()?

Reply:
Umm... IMHO, that's more ambiguous for me. Could you explain more about the strongness when change to `self`?

Reply:
I think I misunderstood. sorry about that.